src/write.jl

```diff
@@ -279,19 +284,26 @@ function write(::NumberType, buf, pos, len, x::AbstractFloat; allow_inf::Bool=fa
     return buf, pos, len
 end
 
-@inline function write(::NumberType, buf, pos, len, x::T; allow_inf::Bool=false, kw...) where {T <: Base.IEEEFloat}
-    isfinite(x) || allow_inf || error("$x not allowed to be written in JSON spec")
-    if isinf(x)
+@inline function write(::NumberType, buf, pos, len, x::T; inf_mapping::Union{Function, Nothing} = nothing, allow_inf::Bool = inf_mapping !== nothing, kw...) where {T <: Base.IEEEFloat}
+    if isfinite(x) || allow_inf && inf_mapping === nothing && isnan(x)
```

Comment:
this mix of `||` and `&&` always makes me nervous because I can't quite remember the order of precedence, so out of an abundance of caution, would you mind adding parens to help make things more clear?

Reply:
Sure